src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -426,15 +426,24 @@ private string GetXrefHref(FilePath file, string uid, int uidCount, bool isRootL
                     return link;
 
                 case JsonSchemaContentType.Markdown:
-
                     // todo: use BuildPage.CreateHtmlContent() when we only validate markdown properties' bookmarks
-                    var html = _markdownEngine.ToHtml(errors, content, sourceInfo, MarkdownPipelineType.Markdown, null, rootSchema.ContentFallback);
-                    return html;
+                    return _markdownEngine.ToHtml(
+                        errors,
+                        content,
+                        sourceInfo,
+                        MarkdownPipelineType.Markdown,
+                        null,
+                        file.Origin == FileOrigin.Fallback || rootSchema.ContentFallback);
 
                 case JsonSchemaContentType.InlineMarkdown:
-
                     // todo: use BuildPage.CreateHtmlContent() when we only validate markdown properties' bookmarks
-                    return _markdownEngine.ToHtml(errors, content, sourceInfo, MarkdownPipelineType.InlineMarkdown, null, rootSchema.ContentFallback);
+                    return _markdownEngine.ToHtml(
+                        errors,
+                        content,
+                        sourceInfo,
+                        MarkdownPipelineType.InlineMarkdown,
+                        null,
```

Comment:
Change LinkResolver.TryResolveRelativePath method instead to:
- Make behavior consistent with markdown include
- Keep the value of contentFallback intact